research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,342 @@
+  "teaching" UI, and other progressive-disclosure widgets or content pickers.
+</p>
+
+## Showing a `popup` element
+
+A `popup` element should not be shown to the user unless:
+
+- The `initiallyopen` attribute is specified
+- The `show()` method is invoked
+- Or the `popup` element has been invoked via the `popup` attribute.
+
+This requirement may be implemented indirectly through the style layer. For example,
+user agents that
+[support the suggested default rendering](https://html.spec.whatwg.org/multipage/infrastructure.html#renderingUA)
+implement this requirement using the CSS rules described in the
+[Rendering section](https://html.spec.whatwg.org/multipage/rendering.html#rendering).
+
+### The `initiallyopen` attribute
+
+The `initiallyopen` attribute is a
+[boolean attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).
+When specified, it indicates that the `popup` element should be shown on document load, such that
+the user can interact with it.
+
+**Example:**
+
+```html
+<popup initiallyopen></popup>
+```
+
+If the `initiallyopen` attribute is specified on a `popup` element _subject_, the user agent must
+run [showing a `popup` element steps](#showing-popup-steps).
+
+<p class="question">
+  When following these run steps, this effectively means the last popup in the DOM order with
+  initiallyopen specified will be the popup that "wins". Is that clear enough to implementers? Is
+  that a reasonable behavior?
+</p>
+
+### The `show()` method
+
+The `show()` method displays the `popup` element, such that the user can interact with it.
+
+**Example:**
+
+```html
+<popup id="newFeatureUI">
+  <p>
+    <strong>New!</strong>
+    I’m some sort of educational UI…
+  </p>
+</popup>
+
+<script>
+  if (upsellNewFeature) {
+    document.getElementById('newFeatureUI').show()
+  }
+</script>
+```
+
+When `show()` is invoked on a `popup` element _subject_, the user agent must
+run [showing a `popup` element steps](#showing-popup-steps).
+
+### The `popup` attribute
+
+The `popup` attribute indicates that a related `popup` element should be shown or hidden as the
+user interacts with the element where the `popup` attribute is specified. If the attribute is
+specified, the attribute's value must be the ID of a `popup` element in the same
+[node document](https://dom.spec.whatwg.org/#concept-node-document).
+
+**Example:** Invoking the `button` element in this example should show/hide the `popup` element,
+depending on whether or not the `popup` element is currently shown:
+
+```html
+<button id="menuButton" popup="menuPopup">Menu</button>
+<popup id="menuPopup" role="menu" anchor="menuButton">
+  <!-- Markup for menuitems goes here -->
+</popup>
+```
+
+The `popup` attribute is supported on a subset of
+[interactive elements](https://html.spec.whatwg.org/multipage/interactive-elements.html):
+
+- `button`
+- `input` in the `button` state
+- `input` in the `text`, `email`, `phone`, or `url` states
```

Comment:
Yeah, I think we should limit to this current set of elements to start. The reasoning is not implementation complexity, it's more about which relationships make semantic sense (a commanding element vs a random div that is not perceivable to the user as interactive).

For `date` and `color`, I don't think we need the `popup` attribute because the controller code should take care of the interaction for the popup part.

Reply:
+1 to a specified list of options for simplicity. What about `number`, `search`, and `password` types?

Reply:
Thanks @una, I just blanked on a few! `search` makes sense, not sure what the use case would be for `number` but that seems ok. `password` we may want to omit for security purposes but I'm not totally sure on that one.